VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -66,16 +72,46 @@ public override Task<SessionHostInfo> CreateAndStart(int instanceNumber, GameRes
                 _logger.LogException($"Failed to start process based host with instance number {instanceNumber}", exception);
                 sessionHostManager.RemoveSessionHost(sessionHostUniqueId);
                 sessionHost = null;
+                string exceptionMessage = exception.ToString();
+                CollectLogs(logFolderPathOnVm, exceptionMessage);
             }
 
             return Task.FromResult(sessionHost);
         }
 
-        public override Task CollectLogs(string processId, string logsFolder, ISessionHostManager sessionHostManager)
+        public override Task CollectLogs(string id, string logsFolder, ISessionHostManager sessionHostManager)
+        {
+            return Task.CompletedTask;
```

Comment:
>return Task.CompletedTas [](http://example.com/codeflow?start=12&length=24)

Curious why we wouldn't return 'NotSupportedException' in this case, similar to what we do in DockerContainerEngine.cs for the other method?

Reply:
It was returning this before this PR, but I can change it to 'NotSupportedException' if that's preferred

Reply:
I think we should be consistent in both scenarios, so let's change to return the exception if possible.